protostar/testing/environments/fuzz_test_execution_environment.py

```diff
@@ -128,6 +147,7 @@ def build_and_run_test(
     ):
         try:
 
+            @multiply_decorator(example, self.produce_examples(function_name))
```

Comment:
I find this a little bit overengineered, because of the trial to generalize the `multiply_decorator` function.

What you are doing here seems to be boiling down to a very trivial reducing for loop:

```python
f = test
for example in reversed(self.state.config.fuzz_examples):
    f = hypothesis.example(**example)(f)

return f
```

this is a very rare use case and I don't see a need to generalise it, and by inlining this you save a lot of code that sets up inputs data structures etc.

(simplified to only handle kwargs, but this is not a point of this suggestion)

Reply:
Not sure why I just wanted to keep this as a decorator with the annotation syntax `@xxx`